sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/CosmosEntityInformation.java

```diff
@@ -493,6 +510,23 @@ private static Field getPartitionKeyField(Class<?> domainType) {
             return partitionKey;
         }
 
+        private static List<String> getTransientFields(Class<?> domainType) {
+            final Field partitionKeyField = getPartitionKeyField(domainType);
+            final List<Field> fields = FieldUtils.getFieldsListWithAnnotation(domainType, Transient.class);
+            List<String> transientFieldNames = new ArrayList<>();
+            Iterator<Field> iterator = fields.iterator();
+            while (iterator.hasNext()) {
+                Field field = iterator.next();
+                if (field.equals(partitionKeyField) || field.getName().equals("id")) {
```

Comment:
what if the field is not named id but has the annotation of id? Where are we checking that?

Reply:
I am not sure that needs checking because it is fine for non id field that is marked `@Id` to be transient. In fact, it might even be preferable in some cases? Worth discussing this as well.

Reply:
Changed as above.